File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        
        if len(s) <10: return
        mp = defaultdict(int) 
        ans = [] 
        for i in range(10,len(s)+1):
            
            subs = s[i-10:i]
            mp[subs] +=1 
            if mp[subs] >=2 and subs not in ans:
                ans.append(subs) 
                
        return ans 
        

File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        
    
    map<string,int> seen ; 
    vector<string> ans ; 
    if(s.size() <10) return ans ; 
    for(int i=0 ; i <= s.size()-9 ; i++){
        //cout << s.substr(i,10) << "\n" ; 
        seen[s.substr(i,10)] ++ ; 
        if( seen[s.substr(i,10)] ==2  ){
            ans.push_back(s.substr(i,10)) ; 
        }
    }
    return ans; 
        
        
    }
};